hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1080,9 +1080,8 @@ void Vgen::emit(const vasm_opc& i) {                         \
   } else {                                                   \
     checkSF(i, StatusFlags::NotV);                           \
     if (!flagRequired(i.fl, StatusFlags::C)) {               \
-      /* Perform the shift and set N and Z. */               \
-      a->arm_opc(gpr_w(i.d), gpr_w(i.s1), i.s0.l());         \
-      a->Bic(vixl::zr, gpr_w(i.d), vixl::zr, SetFlags);      \
+      /* Equivalent to shift and test, set N and Z. */       \
+      a->Tst(gpr_w(i.s1), Operand(~(-1LL << (sz - i.s0.l())))); \
```

Comment:
@jim-saxman - Hi Jim,  this was the case where the actual result of the shift was the status flag.  If you look at the previous sequence the destination of the Bic was vixl::zr.  

I think you might be right in that we cannot know for certain that gpr_w(i.d) is dead.  It may be
that the ones I noticed were only using the status flags.